opensfm/pairs_selection.py

```diff
@@ -531,6 +535,8 @@ def match_candidates_by_time(
 
     pairs = set()
     for image_ref in images_ref:
+        if data.matches_exists(image_ref):
```

Comment:
Have you tested this in a data set of 2 images.

1. Run match
2. Add 3rd image
3. Run match again
4. Observe changes to the matches files?